vmngclient/api/configuration_groups/parcel.py

```diff
@@ -21,32 +17,31 @@ class OptionType(str, Enum):
 
 
 class ParcelValue(BaseModel):
-    class Config:
-        extra = "forbid"
+    model_config = ConfigDict(
+        extra="forbid",
+    )
 
     optionType: OptionType
 
 
-class Global(GenericModel, Generic[T], ParcelValue):
+class Global(Generic[T], ParcelValue):
     optionType: OptionType = OptionType.GLOBAL
     value: T
 
 
-class Variable(GenericModel, ParcelValue):
+# pattern=r'^\{\{[.\/\[\]a-zA-Z0-9_-]+\}\}$', min_length=1, max_length=64
+class Variable(ParcelValue):
     optionType: OptionType = OptionType.VARIABLE
     value: str
 
 
-class Default(GenericModel, Generic[T], ParcelValue):
+class Default(Generic[T], ParcelValue):
     optionType: OptionType = OptionType.DEFAULT
     value: Any
 
 
 class MainParcel(BaseModel):
-    # TODO add constr or Annotated[constr] in Pydantic v2
-    # F722 syntax error in forward annotation
-    # https://github.com/pydantic/pydantic/issues/2872
-    # https://github.com/pydantic/pydantic/issues/156
-    name: str = Field(regex=r'^[^&<>! "]+$', min_length=1, max_length=128)
-    description: Optional[str]
-    data: Parcel
+    # name: Annotated[str, StringConstraints(min_length=1, max_length=128, pattern=r'^[^&<>! "]+$')]
```

Comment:
I believe regular Filed options can be used without using Annotated:
https://github.com/CiscoDevNet/vManage-client/blob/112b1796c8a1beb44c74bf3d9a38888bba2d4371/vmngclient/model/policy/lists_entries.py#L203
regex argument was changed to pattern in V2